src/main/scala/cqrs/domain/orders/Orders.scala

```diff
@@ -42,10 +42,12 @@ class Orders extends PersistentActor with SettingsActor with ActorLogging {
       log.debug("Creating Order")
       persist(OrderCreated(orderId, username)) { evt ⇒
         updateState(evt)
+        orderRegion ! OrderCommandHandler.Command(orderId, Order.InitializeOrder)
```

Comment:
Consider the following scenario:
Message sent to Orders (persisted)
Message sent to Order does not arrive (due to network problem, or persistence problem or something else).

`Orders` knows the `Order`, but the `Order` is not initialized. It is not possible to initialize the order anymore. We need 2 things here: 

1. `Order` needs to be idempotent. Allowing to be initialized multiple times and Acking every time it is initialized. Initializing after it already is initialized, does not really change anything in the `Order`. No event needs to be persisted.
2. `Orders` only persist `OrderCreated` when the `Order` Acknowledge that it is initialized

If we do it this way, we make it fool proof. `Orders` will only know initialized `Order`s.